nixos/thunar: init

romildo opened the pull request:
###### Description of changes

A module for the thunar file manager, which depends on the xfconf dbus service, and also has a dbus service and a systemd unit.

Fixes https://github.com/NixOS/nixpkgs/issues/65771

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

lucasew commented on nixos/modules/programs/thunar.nix:
```diff
@@ -0,0 +1,44 @@
+# Thunar.
```
nit: remove

romildo replied:
Done.

lucasew commented on nixos/modules/programs/thunar.nix:
```diff
@@ -0,0 +1,44 @@
+  };
+
+  config = mkIf cfg.enable (
+    let package = pkgs.xfce.thunar.override { thunarPlugins = cfg.plugins; };
```
nit: break line after let

romildo replied:
Done.

lucasew reviewed (commented):
Minor nitpicks, otherwise LGTM

SuperSamus commented:
Is it appropriate to have both `services.xserver.desktopManager.xfce.thunarPlugins` and `programs.thunar.plugins`?

lucasew commented:
> Is it appropriate to have both `services.xserver.desktopManager.xfce.thunarPlugins` and `programs.thunar.plugins`?

After I posted that review I also realized that, and suggested him to use `mkRenamedOption` in this case.

romildo commented:
> > Is it appropriate to have both `services.xserver.desktopManager.xfce.thunarPlugins` and `programs.thunar.plugins`?
> 
> After I posted that review I also realized that, and suggested him to use `mkRenamedOption` in this case.

Done.

romildo committed:
nixos/thunar: init

- Add a module for the thunar file manager, which depends on the xfconf dbus service, and also has a dbus service and a systemd unit.

- Renames the option services.xserver.desktopManager.xfce.thunarPlugins to programs.thunar.plugins.

jtojnar commented on nixos/modules/programs/thunar.nix:
```diff
@@ -0,0 +1,44 @@
+
+      services.dbus.packages = [
+        package
+        pkgs.xfce.xfconf
```
Might be worth it to have this as a separate module too.

romildo replied:
I may add a module for `xfconf` in another PR.

jtojnar reviewed (approved):
Changes look good, did not test.

lucasew reviewed (approved):
LGTM, didn't test

romildo commented:
I have tested with the Xfce DE and also with the awesome WM.
